src/codegen/function_body_chunk.rs

```diff
@@ -718,13 +718,15 @@ impl Builder {
                 type_: "*mut _".to_string(),
             });
         }
-        finish_args.push(Chunk::Name("res".to_string()));
         finish_args.extend(
             trampoline
                 .output_params
                 .iter()
-                .filter(|param| param.direction == ParameterDirection::Out)
+                .filter(|param| param.direction == ParameterDirection::Out || param.name == "res")
```

Comment:
Don't ever work with the names of parameters :) Use the C type (`GAsyncResult`) or the Rust type `gio::AsyncResult`) here instead.

Async finish functions must have this, and exactly once.

Reply:
Good idea!